### ISSUE TITLE
[ResourceBundle] EventListener\KernelControllerSubscriber should be bug when embed other resource controller.

### ISSUE THREAD

**Issue by liverbool** [Sylius/Sylius#2493]
We cannot embed other `ResourceController` inside view of  `Master ResourceController`, The embed `ResourceController` will override master `parameters`.

```php
    /**
     * @param FilterControllerEvent $event
     */
    public function onKernelController(FilterControllerEvent $event)
    {
       // Master filter should be added here?
       // Not sure about other usecase.
        if (!$event->isMasterRequest()) {
            return;
        }

        .....
    }
```

See:
https://github.com/Sylius/Sylius/blob/master/src/Sylius/Bundle/ResourceBundle/EventListener/KernelControllerSubscriber.php#L74

/cc @aRn0D

**Comment by nmarmon** [comment]
I don't know exactly what do you meanCan you explain it in a more extensive way?

**Comment by jimbocoder** [comment]
I believe this is referring to the `replace` usage, currently located at https://github.com/Sylius/Sylius/blob/master/src/Sylius/Bundle/ResourceBundle/EventListener/KernelControllerSubscriber.php#L96

Rather than an outright replace, probably some sort of recursive merging is necessary, so this subscriber plays nicely with any other subscribers that modify request parameters for the same event.  I had some issues with this myself, and some success experimenting with `array_merge_recursive` but ended up solving my problem a different way and never came up with a decent pull request.

I believe there is a deficiency here, but it's kind of to tell if we're talking about the same thing yet.

**Comment by liverbool** [comment]
The `embed controller (sub-request) --only controller that extended from Sylius/ResourceController` will override master-request's parameters.

**This case happen when:**
  - Master-request is sylius-controller.
  - Embed-request is sylius-controller.

eg.
```yaml
master-request:
  path: /products
  defaults:
    _controller: sylius.controller.product:indexAction
    _sylius:
      sortable: true

sub-request:
  path: /products-filter
  defaults:
    _controller: sylius.controller.product:filterAction
```

http: /products
```twig
{% block content %}
  form.product-filter
       {{ render('sub-request') }}
  
  {# master-reauest.defaults._sylius.sortable will overrided by sub-request. #}
  table
    thead
      tr
        th resource_sort('id', 'ID') {# Now sortable: false #}
         {% for product in products %}
           .....
          {% endfor %}
{% endblock %}
```

I solved this problem with changing embed controller to default symfony controller (none-extended sylius-controller).

**Comment by nmarmon** [comment]
Did you do a pull-request with the solution?

**Comment by liverbool** [comment]
@nmarmon Oops! Nope /:

**Comment by michalmarcinkowski** [comment]
@liverbool this will be fixed after ResourceBundle refactor #2255, thanks for reporting! :+1:
